src/basic/process-util.c

```diff
@@ -675,7 +675,7 @@ int getenv_for_pid(pid_t pid, const char *field, char **_value) {
                 }
                 line[i] = 0;
 
-                if (memcmp(line, field, l) == 0 && line[l] == '=') {
+                if (strneq(line, field, l) && line[l] == '=') {
```

Comment:
@martinpitt but in that case strneq() wouldn't have returned true, and thus the access would not take place...